var/spack/repos/builtin/packages/sim-model/package.py

```diff
@@ -70,11 +70,14 @@ def __build_mods_coreneuron(self, mods_location, link_flag, include_flag):
         spec = self.spec
         assert os.path.isdir(mods_location), mods_location
         include_flag += ' -I%s' % (spec['coreneuron'].prefix.include)
-        which('nrnivmodl-core')(
-            '-i', include_flag, '-l', link_flag, '-n', self.mech_name,
-            '-v', str(spec.version), '-c', mods_location)
+        if spec.satisfies('^coreneuron@0:0.16'):
+            which('nrnivmodl-core')(
+                '-i', include_flag, '-l', link_flag, '-n', self.mech_name,
+                '-v', str(spec.version), '-c', mods_location)
+        else:
+            which('nrnivmodl-core')('-i', include_flag, '-l', link_flag, mods_location)
         output_dir = spec.architecture.target
-        expected_name = "libcorenrnmech" + ('_' + self.mech_name if self.mech_name else '')
+        expected_name = "libcorenrnmech"
```

Comment:
this won't work with `^coreneuron@0:0.16`? because we are still passing `'-v', str(spec.version)` for old version.

Reply:
In the next like we use `expected_name*` as an argument to `find_libraries`, so this function will return anything that starts with `"libcorenrnmech"` which covers both cases

Reply:
@pramodk if this is OK for you can we merge this?

Reply:
yup, sorry for missing this.